stackit/internal/services/iaas/server/resource.go

```diff
@@ -0,0 +1,770 @@
+				Description: "The name of the server.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.LengthAtMost(63),
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[A-Za-z0-9]+((-|\.)[A-Za-z0-9]+)*$`),
+						"must match expression"),
+				},
+			},
+			"machine_type": schema.StringAttribute{
+				Description: "Name of the machine type the server will belong to.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.LengthAtMost(63),
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[A-Za-z0-9]+((-|_|\s|\.)[A-Za-z0-9]+)*$`),
+						"must match expression"),
+				},
+			},
+			"initial_network": schema.SingleNestedAttribute{
+				Description: "The initial networking setup for the server. A network ID or a list of network interfaces IDs can be provided",
+				Optional:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"network_id": schema.StringAttribute{
+						Description: "The network ID",
+						Optional:    true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+						Validators: []validator.String{
+							stringvalidator.ConflictsWith(
+								path.MatchRoot("initial_network").AtName("network_interface_ids"),
+							),
+						},
+					},
+					"network_interface_ids": schema.ListAttribute{
+						ElementType: types.StringType,
+						Description: "List of network interface IDs",
+						Optional:    true,
+						PlanModifiers: []planmodifier.List{
+							listplanmodifier.RequiresReplace(),
+						},
+						Validators: []validator.List{
+							listvalidator.ConflictsWith(
+								path.MatchRoot("initial_network").AtName("network_id"),
+							),
+						},
+					},
+				},
+				Validators: []validator.Object{
+					objectvalidator.AtLeastOneOf(
+						path.MatchRoot("initial_network").AtName("network_id"),
+						path.MatchRoot("initial_network").AtName("network_interface_ids"),
+					),
+				},
+			},
+			"availability_zone": schema.StringAttribute{
+				Description: "The availability zone of the server.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Optional: true,
+				Computed: true,
+			},
+			"boot_volume": schema.SingleNestedAttribute{
+				Description: "The boot volume for the server",
+				Optional:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"delete_on_termination": schema.BoolAttribute{
+						Description: "Delete the volume during the termination of the server. Defaults to `false`.",
+						Optional:    true,
+						PlanModifiers: []planmodifier.Bool{
+							boolplanmodifier.RequiresReplace(),
```

Comment:
If we want to keep it, does it really requires replace? Isn't it some flag we can toggle that will change the behaviour just on deletion?

Reply:
Ah I see it's only used in creation